inspirehep/modules/orcid/utils.py

```diff
@@ -54,3 +60,378 @@ def get_authors_credentials(author):
+        doc_type='authors',
+        body=es_query,
+        size=1
+    )['hits']['hits']
+    return records
+
+
+def query_for_author_profiles_using_full_name(authors_full_name):
+    es_query = {
+        "query": {
+            "bool": {
+                "must": [{
+                    "match": {
+                        "name.name_variations": authors_full_name
+                    }
+                }
+                ],
+                "should": [{
+                    "match": {
+                        "name.preferred_name": authors_full_name
+                    }},
+                    {
+                    "match": {
+                        "name.value": authors_full_name
+                    }
+                }
+                ]
+            }
+        }
+    }
+    records = current_search_client.search(
+        index='records-authors',
+        doc_type='authors',
+        body=es_query,
+        size=10000
+    )['hits']['hits']
+    return records
+
+
+def get_identifiers_from_orcid_records(orcid, token):
+
+    api = current_app.extensions['inspire-orcid'].orcid_api
+
+    orcid_records = api.read_record_public(
+        orcid, 'activities', token)
+    orcid_records_dois = []
+    orcid_records_arxivs = []
+    for record in orcid_records['works']['group']:
+        if record['external-ids']['external-id'][0]['external-id-type'] == 'doi':
+            orcid_records_dois.append(
+                record['external-ids']['external-id'][0]['external-id-value'])
+        if record['external-ids']['external-id'][0]['external-id-type'] == 'arxiv':
+            orcid_records_arxivs.append(
+                record['external-ids']['external-id'][0]['external-id-value'])
+
+    return orcid_records_dois, orcid_records_arxivs
+
+
+def extract_author_profiles_from_hep_records(records):
+    possible_profiles = set()
+    for record in records:
+        try:
+            for author in record['inner_hits']['authors']['hits']['hits']:
+                possible_profiles.add(author['_source']['record']['$ref'])
+        except KeyError:
+            continue
+    return possible_profiles
+
+
+def solve_references_of_authors(refs):
+    possible_profiles = []
+    current_app.logger.info(refs)
+    for ref in refs:
+        author_profile = replace_refs({'$ref': ref}, source='es')
+        try:
+            # If the profile has an orcid it's already claimed by somebody.
+            for identifier in author_profile['ids']:
+                if 'orcid' in identifier:
+                    continue
+            possible_profiles.append(ref)
+        except KeyError:
+            continue
+    return possible_profiles
+
+
+def pick_records_from_author_profiles(profiles):
+    number_of_suggested_records = len(profiles)
+    number_of_records_per_profile = 1
+    picked_records = {}
+    picked_recids = []
+    if len(profiles) is 2:
+        number_of_suggested_records = 4
+        number_of_records_per_profile = 2
```

Comment:
I see, I think it will work although it was a bit difficult to read. Maybe adding a test will improve that ;)